Source/WebKit/NetworkProcess/NetworkProcess.cpp

```diff
@@ -1525,10 +1525,10 @@ void NetworkProcess::preconnectTo(PAL::SessionID sessionID, WebPageProxyIdentifi
 
     NetworkLoadParameters parametersForAdditionalPreconnect = parameters;
 
-    session->networkLoadScheduler().startedPreconnectForMainResource(url, userAgent);
+    session->protectedNetworkLoadScheduler()->startedPreconnectForMainResource(url, userAgent);
     auto task = new PreconnectTask(*session, WTFMove(parameters), [session = WeakPtr { *session }, url, userAgent, parametersForAdditionalPreconnect = WTFMove(parametersForAdditionalPreconnect)](const WebCore::ResourceError& error, const WebCore::NetworkLoadMetrics& metrics) mutable {
-        if (session) {
-            session->networkLoadScheduler().finishedPreconnectForMainResource(url, userAgent, error);
+        if (CheckedPtr checkedSession = session.get()) {
```

Comment:
I would have renamed the captured variable to `weakSession` and used `session` for this variable instead. Make it clearer people are dealing with a weak ptr